File: src/app/models/user.model.ts
export interface User {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
  avatar: string;
  role: string;
  completedTasks: number;
  totalPoints: number;
  createdAt: Date;
}

export interface AuthResponse {
  token: string;
  userId: string;
  email: string;
  firstName: string;
  lastName: string;
  avatar: string;
  role: string;
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface RegisterRequest {
  email: string;
  password: string;
  firstName: string;
  lastName: string;
  avatar: string;
  role: string;
}

File: API/HO.FamilyTicketTracker/HO.FamilyTicketTracker.API/Models/DTOs/UserModel.cs
namespace HO.FamilyTicketTracker.API.Models.DTOs
{
  public class UserModel
  {
    public string Id { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Avatar { get; set; } = string.Empty;
    public string Role { get; set; }
    public int CompletedTasks { get; set; }
    public int TotalPoints { get; set; }
    public DateTime CreatedAt { get; set; }
  }

  public static class UserModelExtensions
  {
    public static UserModel ToUserModel(this User user)
    {
      return new UserModel
      {
        Id = user.Id,
        Email = user.Email ?? string.Empty,
        FirstName = user.FirstName,
        LastName = user.LastName,
        Avatar = user.Avatar,
        Role = user.Role,
        CompletedTasks = user.CompletedTasks,
        TotalPoints = user.TotalPoints,
        CreatedAt = user.CreatedAt
      };
    }

    public static User ToUser(this UserModel user)
    {
      return new User
      {
        Id = user.Id,
        Email = user.Email ?? string.Empty,
        FirstName = user.FirstName,
        LastName = user.LastName,
        Avatar = user.Avatar,
        Role = user.Role,
        CompletedTasks = user.CompletedTasks,
        TotalPoints = user.TotalPoints,
        CreatedAt = user.CreatedAt
      };
    }
  }
}
